src/main/typescript/data/common.ts

```diff
@@ -311,3 +300,74 @@ export class CompositeResource extends CompositeNumber implements Resource {
     return clone;
   }
 }
+
+/**
+ * By default, zod strips unknown keys. This is not the default desired
+ * behaviour here. This is a thin wrapper.
+ * @param shape - The desired object shape
+ * @returns The desired zod object
+ */
+export function zObject<T extends z.ZodRawShape>(
+  shape: T
+): z.ZodObject<T, "strict"> {
+  return z.object(shape).strict();
+}
```

Comment:
Biggest reason I did it this way was because it would have taken _ages_ to manually add `.strict()` at the end of all the types. Here I was able to just search for `z.object` and replace it.
If you think it's really superfluous, I can manually go through each occurence and add `.strict()`

Reply:
Yes I would prefer that.

Reply:
Done. Current solution feels a bit hacky tho, imo